src/Coordination/Changelog.cpp

```diff
@@ -280,16 +296,10 @@ Changelog::Changelog(
 
 void Changelog::readChangelogAndInitWriter(uint64_t last_commited_log_index, uint64_t logs_to_keep)
 {
-    uint64_t total_read = 0;
-
-    /// Amount of entries in last log index
-    uint64_t entries_in_last = 0;
-    /// Log idx of the first incomplete log (key in existing_changelogs)
-    int64_t first_incomplete_log_start_index = -1; /// if -1 then no incomplete log exists
+    std::optional<ChangelogReadResult> last_log_read_result;
 
-    ChangelogReadResult result{};
-    /// First log index which was read from all changelogs
-    uint64_t first_read_index = 0;
+    /// Last log has some free space to write
+    bool last_log_is_not_complete = false;
 
     /// We must start to read from this log index
     uint64_t start_to_read_from = last_commited_log_index;
```

Comment:
Is `last_commited_log_index == 0` is valid index? Does it mean we have one record on nothing?

Reply:
It's valid and it means that we have nothing.